File: sdks/java/src/main/java/com/boltprediction/sdk/model/LeakageParamsSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 泄漏率估算模型参数 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class LeakageParamsSchema {

    @JsonProperty("base_leakage_rate_m3_per_hour")
    private Double baseLeakageRateM3PerHour;

    @JsonProperty("critical_leakage_threshold")
    private Double criticalLeakageThreshold;

    @JsonProperty("preload_leakage_sensitivity")
    private Double preloadLeakageSensitivity;

    @JsonProperty("seal_aging_factor_per_year")
    private Double sealAgingFactorPerYear;

    @JsonProperty("pressure_sensitivity")
    private Double pressureSensitivity;

    public Double getBaseLeakageRateM3PerHour() {
        return baseLeakageRateM3PerHour;
    }

    public void setBaseLeakageRateM3PerHour(Double baseLeakageRateM3PerHour) {
        this.baseLeakageRateM3PerHour = baseLeakageRateM3PerHour;
    }

    public Double getCriticalLeakageThreshold() {
        return criticalLeakageThreshold;
    }

    public void setCriticalLeakageThreshold(Double criticalLeakageThreshold) {
        this.criticalLeakageThreshold = criticalLeakageThreshold;
    }

    public Double getPreloadLeakageSensitivity() {
        return preloadLeakageSensitivity;
    }

    public void setPreloadLeakageSensitivity(Double preloadLeakageSensitivity) {
        this.preloadLeakageSensitivity = preloadLeakageSensitivity;
    }

    public Double getSealAgingFactorPerYear() {
        return sealAgingFactorPerYear;
    }

    public void setSealAgingFactorPerYear(Double sealAgingFactorPerYear) {
        this.sealAgingFactorPerYear = sealAgingFactorPerYear;
    }

    public Double getPressureSensitivity() {
        return pressureSensitivity;
    }

    public void setPressureSensitivity(Double pressureSensitivity) {
        this.pressureSensitivity = pressureSensitivity;
    }

}

File: sdks/python/bolt_prediction_sdk/models/leakage_params_schema.py
"""
LeakageParamsSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class LeakageParamsSchema(SDKBaseModel):
    """泄漏率估算模型参数"""

    base_leakage_rate_m3_per_hour: Optional[float] = Field(description="基准泄漏率 (m³/h)", default=0.0)
    critical_leakage_threshold: Optional[float] = Field(description="临界泄漏压紧比阈值", default=0.05)
    preload_leakage_sensitivity: Optional[float] = Field(description="预紧力泄漏敏感度指数", default=2.5)
    seal_aging_factor_per_year: Optional[float] = Field(description="密封年老化系数", default=0.08)
    pressure_sensitivity: Optional[float] = Field(description="压力敏感度", default=1.2)
